src/dt/perspectives/stereotype/bias_generation.py

```diff
@@ -329,8 +288,47 @@ def main(OPTS):
             print("#/ prompt tokens:", prompt_tokens)
             print("#/ completion tokens:", completion_tokens)
             if dry_run == False:
-                calculate_agreeability(OPTS.sys_prompt_type, model_name, n_generations, csv_prefix)
+                calculate_agreeability(OPTS.sys_prompt_type, model_name, n_generations, org_name)
     else:
-        org_name = model_name.split('/')[0]
-        model_name = model_name.strip(org_name + '/')
+        org_name, model_name = name_handler(OPTS.model)
         calculate_agreeability(OPTS.sys_prompt_type, model_name, n_generations, org_name)
+
+        #assumes generations have already been made for all the three system prompt types: benign, untargeted, and targeted
+        try:
+            create_json(org_name, model_name)
+        except:
+            print("Skipping compiled JSON generation. Execute all the perspectives (benign, untargeted, targeted) for the JSON to be generated.")
+
+        # old saving format
+        #org_name = model_name.split('/')[0]
+        #model_name = model_name.strip(org_name + '/')
+        #calculate_agreeability(OPTS.sys_prompt_type, model_name, n_generations, org_name)
+
+def make_directories(model_full_name):
+
+    org_name, model_name = name_handler(model_full_name)
+
+    if not os.path.exists('./data/stereotype/'):
+        os.mkdir('./data/stereotype/')
+
+    if not os.path.exists('./data/stereotype/generations/'):
+        os.mkdir('./data/stereotype/generations/')
+
+    if not os.path.exists('./data/stereotype/generations/' + org_name):
+        os.mkdir('./data/stereotype/generations/' + org_name)
+
+    if not os.path.exists('./data/stereotype/outputs/'):
+        os.mkdir('./data/stereotype/outputs/')
+    
+    if not os.path.exists('./data/stereotype/outputs/' + org_name):
+        os.mkdir('./data/stereotype/outputs/' + org_name)
```

Comment:
@garyxcj Did we refactor the outputs directory of `stereotype` before?

Reply:
@ritik99 Can we refactor this into `./results/stereotype`? Also, try not to hardcode your saving paths. Determine the save path programmatically or use variable interpolation in the configuration file.

Reply:
resolved in [7f8af8b](https://github.com/AI-secure/DecodingTrust/pull/8/commits/7f8af8ba64d0dfcbe8d5e3c8607820aada05d099)